internal/release/cross/release_list.go

```diff
@@ -160,6 +163,25 @@ func (r *ReleaseList) GetReleasesForPromotion(sourceEnv, targetEnv *v1alpha1.Env
 	return subset, nil
 }
 
+// GetNonPromotableReleases returns a list of names of releases that cannot be promoted based
+// on the version format allowed at the target environment.
+func (r *ReleaseList) GetNonPromotableReleases(sourceEnv, targetEnv *v1alpha1.Environment) []string {
+	var invalidList []string
+	sourceEnvIndex := r.getEnvironmentIndex(sourceEnv.Name)
+
+	for _, item := range r.Items {
+		sourceRelease := item.Releases[sourceEnvIndex]
+		// Check the version format in source Release
+		if !targetEnv.Spec.Promotion.FromPullRequests {
```

Comment:
It would be clearer to the reader if you can perform this check outside of the loop. If the environment accepts PullRequests you don't need to perform this loop.

```go
if sourceEnv.Spec.Promotion.FromPullRequests { 
  return nil
}
```
```